### PR TITLE
Fix Button styling details

### PR DESCRIPTION
**Pull request by alimpens**

### CODE REVIEW

File: packages/css/src/button/button.scss [hunk 1]
```diff
@@ -14,6 +14,38 @@
...
+// Specificity hack to be able to style secondary button focus state
+// This isn't currently possible with the Utrecht component
+// We should really pull this component in our own repo...
+.amsterdam-button--secondary.amsterdam-button--secondary {
+  &:focus {
+    color: var(--amsterdam-button-secondary-focus-color);
+  }
+}
+
+.amsterdam-button--secondary:focus:not(:disabled, [aria-disabled="true"]) {
```
Comment (VincentSmedinga) [inline packages/css/src/button/button.scss:39]: Should we include `.utrecht-button--disabled` in lists like this?

File: packages/css/src/button/button.scss [hunk 2]
```diff
@@ -14,6 +14,38 @@
   }
 }
 
-.amsterdam-button--secondary:hover {
-  box-shadow: var(--amsterdam-button-secondary-hover-box-shadow);
+.amsterdam-button--secondary {
+  box-shadow: var(--amsterdam-button-secondary-box-shadow);
+
+  &:disabled,
```
[the remark this answers is not in the document]
Comment (alimpens) [reply packages/css/src/button/button.scss:20]: Setting `pointer-events: none` for the `disabled` state also overrides the `cursor: not-allowed` rule for this state.

The `disabled` prop also prevents JavaScript events from firing (in React at least, not sure about vanille js), so there's no difference in functionality there. 

I think I prefer changing the cursor for a disabled state vs less css, what do you think?

File: packages/css/src/button/button.scss (hunk 1 above)
Comment (alimpens) [reply packages/css/src/button/button.scss:39]: I don't think it's necessary, the Utrecht Button component doesn't set a `box-shadow`, so everything `box-shadow`-related is contained within the `amsterdam-button` namespace

File: packages/css/src/button/button.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages/css/src/button/button.scss:20]: I’ll have to agree, user more important than code. Let’s keep an eye on the error-proneness.

**Review by VincentSmedinga — APPROVED** [review]

**Pull request merged**